### ISSUE TITLE
Explain Consumer performance optimizations. 

### ISSUE THREAD

**Issue by feinstein** [rrousselGit/provider#172]
The README.md docs say `Consumer` can be used for performance optimizations, but don't explain what are these optimizations. I think it refers to using the `child` property, but I only learned about it in other tutorials. I think the docs should be improved from a beginner's perspective.

**Comment by anticafe** [comment]
In fact I still don't understand what's the different between using `Consumer` vs. without `Consumer`.

**Comment by feinstein** [comment]
As far as I understood, `Consumer` can provide a `child` along with the value. This child is a `Widget` tree that you made before and gave it to `Consumer`. This ʼWidget` tree is sopposed to be immutable, so you only create it once, and reuse it, so if the value is updated only the `Widgets` that need to be updated are reconstructed.

I think an example will be more clear that my explanation, hence the request for better documentations. 

Also, there may be other differences that I am not aware of.

**Comment by rrousselGit** [comment]
Ah, actually I explained that in a recent talk.
I'll add it to the documentation soon.

TL;DR: Consumer vs Provider.of changes the widget that rebuilds. 

With Provider.of, it's the widget that called that method which rebuilds. With Consumer, it's the Consumer. 

That combined with the optional `child` allows for more gralunar rebuild of the widget tree.

**Comment by feinstein** [comment]
When you say `With Provider.of, it's the widget that called that method which rebuilds`, you mean that the `Widget` that we are coding the `build` method, the one that we got `context` from, will be rebuild, right? And not the `Widget` that we placed `Provider.of` inside, like inside a `RaisedButton` `onPressed` method in the middle of the `build` method.

**Comment by rrousselGit** [comment]
> you mean that the Widget that we are coding the build method, the one that we got context from, will be rebuild, right?

Yes. It's the widget associated with the `BuildContext` passed to `Provider.of` that will rebuild when the value changes.

We could technically use `Builder` instead of `Consumer`. It'd do the same thing.

**Comment by feinstein** [comment]
It will be nice to have an example at the docs, showing who will get rebuild, as the technical description can be ambiguous for begginers.

**Comment by rrousselGit** [comment]
https://github.com/rrousselGit/provider/pull/176

I opened a PR that improves the documentation on `Consumer`. Is that enough?

**Comment by rrousselGit** [comment]
> Also, does it speaks about dispose?

What dispose?

**Comment by feinstein** [comment]
The dispose method for when the provider goes away and we can destroy our resources.

**Comment by feinstein** [comment]
Oh sorry, that's on the provider, here we are at the other end, the consumer, my bad, I am trying to walk and contribute at the same time 😅

**Comment by feinstein** [comment]
One thing more I would add is an article summing it all, as reading class documentations only help if you already knows how the library works and you know which classes are you looking for.
